modelforge/potential/models.py

```diff
@@ -29,54 +39,77 @@ def __init__(self, cutoff: float = 5.0):
         self.calculate_neighbors = neighbor_pairs_nopbc
         self.cutoff = cutoff
 
-    def compute_r_ij(self, atom_index12: torch.Tensor, R: torch.Tensor) -> torch.Tensor:
+    def compute_r_ij(
+        self, atom_pairs: torch.Tensor, positions: torch.Tensor
+    ) -> torch.Tensor:
         """Compute displacement vector between atom pairs.
 
         Parameters
         ----------
-        atom_index12 : torch.Tensor, shape [n_pairs, 2]
+        atom_pairs : torch.Tensor, shape [n_pairs, 2]
             Atom indices for pairs of atoms
-        R : torch.Tensor, shape [batch_size, n_atoms, 3]
+        positions : torch.Tensor, shape [batch_size, n_atoms, 3]
             Atom coordinates.
 
         Returns
         -------
         torch.Tensor, shape [n_pairs, 3]
             Displacement vector between atom pairs.
         """
-        coordinates = R.flatten(0, 1)
-        selected_coordinates = coordinates.index_select(0, atom_index12.view(-1)).view(
+        coordinates = positions.flatten(0, 1)
+        selected_coordinates = coordinates.index_select(0, atom_pairs.view(-1)).view(
```

Comment:
I don't think the logic here is correct. 

let us assume we have a positions tensor shape [2,2,3], so two two batches, each with 2 atoms (with one pair between the two atoms in each batch):

`
positions = torch.tensor([[[0. , 0. , 0. ],[1. , 0. , 0. ]],[[3. , 0. , 0. ],[3. , 0.5, 0. ]]])
atom_pairs = torch.tensor([[0, 1],[2, 3]])
`
If we apply the logic in the code, selected_coordinates ends up with:

`
tensor([[[0.0000, 0.0000, 0.0000],
         [1.0000, 0.0000, 0.0000]],

        [[3.0000, 0.0000, 0.0000],
         [3.0000, 0.5000, 0.0000]]])
`
so therefore the the difference (selected_coordinates[0]-selected_coordinates[1]) ends up:

`
tensor([[-3.0000,  0.0000,  0.0000],
        [-2.0000, -0.5000,  0.0000]])
`
Obviously the difference between atoms in two batches, not within the same batch. 

I think line 60 should be:

`
selected_coordinates = coordinates.index_select(0, pairs.view(-1)).view(2,-1).split(3,-1)
`

for this selected_coordinates are:

`
(tensor([[0., 0., 0.],
         [3., 0., 0.]]),
 tensor([[1.0000, 0.0000, 0.0000],
         [3.0000, 0.5000, 0.0000]]))
`

and thus the difference is what we'd expect from the input data.  

`
tensor([[-1.0000,  0.0000,  0.0000],
        [ 0.0000, -0.5000,  0.0000]])
`

Reply:
This is not incorrect; the issue stems from the order of indices in the atom_pairs.  It is NOT `[[a_i, a_j], [b_i, b_j]]`, but rather `[[a_i, b_i], [a_j, b_j]]` (where `a` is the first batch, `b` is the second batch, for pairs `i` and `j`).   It appears the docstring is not correct.  Should be shape [2, n_pairs], where atom_pairs[0] lists all of the index `i` atoms in the pair  , atom_pairs[1]  refers to the `j` .

Reply:
Docstring has been updated so this has been resolved.